pkg/stash/with_redis_test.go

```diff
@@ -49,6 +50,63 @@ func TestWithRedisLock(t *testing.T) {
 	}
 }
 
+// Verify with WithRedisLock working with password protected redis
+// Same logic as the TestWithRedisLock test.
+func TestWithRedisLockWithPassword(t *testing.T) {
+	endpoint := os.Getenv("REDIS_TEST_ENDPOINT2")
+	password := os.Getenv("ATHENS_REDIS_PASSWORD2")
+	if len(endpoint) == 0 {
+		t.SkipNow()
+	}
+	strg, err := mem.NewStorage()
+	if err != nil {
+		t.Fatal(err)
+	}
+	ms := &mockRedisStasher{strg: strg}
+	wrapper, err := WithRedisLock(endpoint, password, strg)
+	if err != nil {
+		t.Fatal(err)
+	}
+	s := wrapper(ms)
+
+	var eg errgroup.Group
+	for i := 0; i < 5; i++ {
+		eg.Go(func() error {
+			ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+			defer cancel()
+			_, err := s.Stash(ctx, "mod", "ver")
+			return err
+		})
+	}
+
+	err = eg.Wait()
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// Verify the WithRedisLock fails with the correct error when trying
+// to connect with the wrong password.
+func TestWithRedisLockWithWrongPassword(t *testing.T) {
+	endpoint := os.Getenv("REDIS_TEST_ENDPOINT2")
+	password := ""
+	if len(endpoint) == 0 {
+		t.SkipNow()
+	}
+	strg, err := mem.NewStorage()
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = WithRedisLock(endpoint, password, strg)
+	if err == nil {
+		t.Fatal("Expected Connection Error")
+	}
+
+	if err.Error() != "NOAUTH Authentication required." {
```

Comment:
Is there a Redis guarantee that this message will always stay the same? Maybe we can type assert to whatever client error and check something more concrete like an error number or a specific error type.

Reply:
This is a good idea, let me check if we can do something other `string matching`

Reply:
looks like there's no guaranteed error specified in the docs either, so +1 from me on string matching

Reply:
Yeah i double checked. It has a strange `cmd` handler for the result as well.

Reply:
Unfortunate, but that's what we're left with. I don't think string matching will be materially better unfortunately.

If the test breaks because the error message changed, I don't think this will be that hard to track down.